tests/ui/pattern/usefulness/issue-30240.stderr

```diff
@@ -18,6 +19,7 @@ LL |     match "world" {
    |           ^^^^^^^ pattern `&_` not covered
    |
    = note: the matched value is of type `&str`
+   = note: `str` cannot be matched exhaustively, so a wildcard `_` is necessary
```

Comment:
Instead of adding an additional note, I wonder if it's easier to just specialize the help message "ensure that all possible cases are being handled by adding a match arm with a wildcard pattern or an explicit pattern as shown" below to be a bit more *context specific*, explaining something a bit more directly about how strings aren't exhaustive.

Reply:
I would also prefer we didn't add a note, but I don't know how we'd phrase it. The tricky thing is that the non exhaustive type may be deep in a pattern somewhere (e.g. `Result<Option<&str>, ...>`), which prevents us from saying e.g. "ensure all possible strings are being handled".

Reply:
Oh oops forgot about this one before merging. I think it's ok like this, if one of you has a better idea we can make another PR. My secret plan is to suggest actual missing strings, something like `patterns Some(""), Some("a"), Some("b") and many others not covered` but I expect that will take some work.